src/library_exceptions.js

```diff
@@ -131,6 +131,23 @@ var LibraryExceptions = {
     };
   },
 
+  $CppException__docs: '/** @constructor */',
+  $CppException__deps: [
+#if !DISABLE_EXCEPTION_CATCHING
+    '$getExceptionMessage',
+#endif
+    '$EmscriptenEH'],
+  $CppException: 'function CppException(excPtr) {\n' +
+                 '    this.name = \'CppException\';\n' + 
+#if DISABLE_EXCEPTION_CATCHING
+                 '    this.message = excPtr;\n' +
+#else
+                 '    this.message = getExceptionMessage(excPtr);\n' +
+#endif
+                 '    this.stack = Error().stack;\n' +
+                 '}\n' +
+                 'CppException.prototype = new EmscriptenEH;',
```

Comment:
This creates an error with the stack at the time of creation. It should be `= EmscriptenEH.prototype` instead. Better yet, let's just use ES6 classes - Emscripten already uses ES6 in other places, and this would make the code both more natural and fixup the inheritance.

Reply:
b0bb3f60597a358ca00fe0cb92cb9d663d14ff2a